### ISSUE TITLE
feat: implement `set_int` function `data_store` contract.

### ISSUE THREAD

**Issue by sparqet** [keep-starknet-strange/satoru#59]
## Description

Implement the `set_int` function in the `data_store` contract.

File: `src/data/data_store.cairo`.

Function signature:
```rust
    /// Set the int value for the given key.
    /// # Arguments
    /// `key` - The key of the value
    /// `value` - The value to set
    /// # Return
    /// The int value for the key.
    fn set_int(ref self: TContractState, key: felt252, value: i128) -> i128;
```

## Acceptance criteria

- Function is implemented and well commented.
- Associated unit tests (nominal and failure) are implemented and passing.

## Resources

- [setInt function in GMX repo](https://github.com/gmx-io/gmx-synthetics/blob/77a2ff39f1414a105e8589d622bdb09ac3dd97d8/contracts/data/DataStore.sol#L151)

**Comment by danilowhk** [comment]
Can start with this :)

**Comment by abdelhamidbakhta** [comment]
> Can start with this :)

assigned to you

**Comment by danilowhk** [comment]
## Currently trying to implement it and am getting these errors:

```cairo

error: Trait has no implementation in context: core::serde::Serde::<core::integer::i128>
 --> IDataStoreDispatcherTrait:2584:39
                serde::Serde::<i128>::deserialize(ref ret_data),
                                      ^*********^

error: Trait has no implementation in context: core::serde::Serde::<core::integer::i128>
 --> IDataStoreDispatcherTrait:2599:31
        serde::Serde::<i128>::serialize(@value, ref __calldata__);
                              ^*******^

error: Trait has no implementation in context: core::serde::Serde::<core::integer::i128>
 --> contract:902:27
    serde::Serde::<i128>::serialize(@res, ref arr);
                          ^*******^

error: Trait has no implementation in context: core::serde::Serde::<core::integer::i128>
 --> contract:916:31
        serde::Serde::<i128>::deserialize(ref data),
                              ^*********^


error: Trait has no implementation in context: core::starknet::storage_access::Store::<core::integer::i128>
 --> contract:281:46
                    starknet::Store::<i128>::read(
                                             ^**^

error: Trait has no implementation in context: core::starknet::storage_access::Store::<core::integer::i128>
 --> contract:291:46
                    starknet::Store::<i128>::write(
                                             ^***^


```

## For i128 Serde:
### It implemented a few days ago, but not available on the current latest scarb version 0.7.0.

https://github.com/starkware-libs/cairo/blob/7d66e9ad67159d37115b89b985d1b9e030f1c863/corelib/src/integer.cairo#L2316

## For i128 Store:
### it seems there is no implementation yet:

https://github.com/starkware-libs/cairo/blob/main/corelib/src/starknet/storage_access.cairo

## I have tried implementing them locally:

```cairo
impl I128Serde of Serde<i128> {
    fn serialize(self: @i128, ref output: Array<felt252>) {
        Into::<i128, felt252>::into(*self).serialize(ref output);
    }
    fn deserialize(ref serialized: Span<felt252>) -> Option<i128> {
        Option::Some(((*serialized.pop_front()?).try_into())?)
    }
}

impl StoreI128 of Store<i128> {
    fn read(address_domain: u32, base: StorageBaseAddress) -> SyscallResult<i128> {
        Result::Ok(
            Store::<felt252>::read(address_domain, base)?.try_into().expect('StoreI128 - non i128')
        )
    }
    #[inline(always)]
    fn write(address_domain: u32, base: StorageBaseAddress, value: i128) -> SyscallResult<()> {
        Store::<felt252>::write(address_domain, base, value.into())
    }
    #[inline(always)]
    fn read_at_offset(
        address_domain: u32, base: StorageBaseAddress, offset: u8
    ) -> SyscallResult<i128> {
        Result::Ok(
            Store::<felt252>::read_at_offset(address_domain, base, offset)?
                .try_into()
                .expect('StoreI128 - non i128')
        )
    }
    #[inline(always)]
    fn write_at_offset(
        address_domain: u32, base: StorageBaseAddress, offset: u8, value: i128
    ) -> SyscallResult<()> {
        Store::<felt252>::write_at_offset(address_domain, base, offset, value.into())
    }
    #[inline(always)]
    fn size() -> u8 {
        1_u8
    }
}

```

### But then still getting these errors:

```cairo
error: Trait has no implementation in context: core::serde::Serde::<core::integer::i128>
 --> contract:902:27
    serde::Serde::<i128>::serialize(@res, ref arr);
                          ^*******^

error: Trait has no implementation in context: core::serde::Serde::<core::integer::i128>
 --> contract:916:31
        serde::Serde::<i128>::deserialize(ref data),
                              ^*********^


error: Trait has no implementation in context: core::starknet::storage_access::Store::<core::integer::i128>
 --> contract:281:46
                    starknet::Store::<i128>::read(
                                             ^**^

error: Trait has no implementation in context: core::starknet::storage_access::Store::<core::integer::i128>
 --> contract:291:46
                    starknet::Store::<i128>::write(
                                             ^***^
```

### Wondering if I am implementing it incorrectly or its just because the implementation needs to com from corelibs
